src/include/units/random.h

```diff
@@ -0,0 +1,453 @@
+#pragma once
+
+#include <random>
+#include <functional>
+
+namespace units {
+
+template<typename Quantity>
+struct uniform_int_distribution : public std::uniform_int_distribution<typename Quantity::rep>
+{
+    using Rep = Quantity::rep;
+    using Base = std::uniform_int_distribution<Rep>;
+    
+    uniform_int_distribution() : Base() {}
+    uniform_int_distribution(Quantity a, Quantity b) : Base(a.count(), b.count()) {}
```

Comment:
Pass a quantity by reference (representation type can be big)

Reply:
Should ``rep``` also be passed by reference?

Reply:
Yes, `rep` is the only content of the quantity and it might be a big type. This is why the `quantity` constructor looks as follows:

https://github.com/mpusz/units/blob/5dd9eaac87a50adc62b170987ebac2d1344d63c5/src/include/units/quantity.h#L76-L78